src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -21,9 +21,24 @@ internal class Command
         private readonly StreamForwarder _stdOut;
         private readonly StreamForwarder _stdErr;
 
+        public enum CommandResolutionStrategy
```

Comment:
Strategy seems right. The result would include the path & args for the command, as well as the strategy that selected that command.